src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -0,0 +1,547 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import htsjdk.variant.vcf.VCFSimpleHeaderLine;
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.InfiniteRandomMatingPopulationModel;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.PloidyModel;
+import org.broadinstitute.hellbender.utils.*;
+import org.broadinstitute.hellbender.utils.genotyper.ReadLikelihoods;
+import org.broadinstitute.hellbender.utils.genotyper.SampleList;
+import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
+import org.broadinstitute.hellbender.utils.locusiterator.LocusIteratorByState;
+import org.broadinstitute.hellbender.utils.pileup.PileupElement;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+import org.broadinstitute.hellbender.utils.read.AlignmentUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * Code for estimating the reference confidence
+ *
+ * This code can estimate the probability that the data for a single sample is consistent with a
+ * well-determined REF/REF diploid genotype.
+ *
+ */
+public final class ReferenceConfidenceModel {
+
+    private final GenomeLocParser genomeLocParser;
+
+    private final SampleList samples;
+    private final int indelInformativeDepthIndelSize;
+
+    private static final boolean WRITE_DEBUGGING_BAM = false;
```

Comment:
Is is useful to keep this debugging stuff? is it actually ever used? Sorry, I don't know the answer.